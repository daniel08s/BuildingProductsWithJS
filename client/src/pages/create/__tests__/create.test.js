/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import CreatePage, {Create} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
  registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
};

test('# CreatePage', () => {
  const store = mockStore({auth: {user}});

  const wrapper = shallow(<CreatePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Create', () => {
  const newText = 'This is a test question, right?';
  // const newExpirationDate = new Date(2018, 1, 1, 1, 1, 1, 1);
  const doCreateQuestion = ({text}) => { // TODO: add expiration date
    expect(text).toBe(newText);
    // expect(expirationDate).toBe(newExpirationDate.toISOString());
  };

  const component = (
    <Create
      user={user}
      doCreateQuestion={doCreateQuestion}
    />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test interaction
  const app = mount(component);

  /* Test Create Question */
  app.find('#questionText').getDOMNode().value = newText;
  // app.find('#expirationDate').getDOMNode().value = newExpirationDate.toISOString();

  // click create button
  app.find('button').simulate('click');
});
