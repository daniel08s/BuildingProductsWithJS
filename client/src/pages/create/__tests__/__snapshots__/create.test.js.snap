// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Create 1`] = `
<div>
  <Component
    current="/create"
    user={
      Object {
        "id": "1",
        "login": "testuser",
        "registrationDate": 2018-02-01T01:01:01.001Z,
      }
    }
  />
  <div
    className="container"
  >
    <form
      className="formQuestion"
    >
      <div
        className="form-group"
      >
        <p
          className="h4 mb-3 font-weight-normal"
        >
          Please fill the form to create a new question
        </p>
        <label
          className="sr-only"
          htmlFor="questionText"
        />
        <textarea
          className="form-control"
          id="questionText"
          placeholder="Question text"
          required={true}
          rows="5"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="sr-only"
          htmlFor="expirationDate"
        />
        <input
          className="form-control"
          id="expirationDate"
          placeholder="Expiration Date"
          required={true}
          type="date"
        />
      </div>
      <p />
      <button
        className="btn btn-lg btn-primary btn-block"
        onClick={[Function]}
        type="submit"
      >
        Create new question
      </button>
    </form>
  </div>
</div>
`;

exports[`# CreatePage 1`] = `
<Create
  doCreateQuestion={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={
    Object {
      "id": "1",
      "login": "testuser",
      "registrationDate": 2018-02-01T01:01:01.001Z,
    }
  }
/>
`;
